eva-accession-pipeline/src/main/java/uk/ac/ebi/eva/accession/pipeline/steps/tasklets/reportCheck/ReportCheckTasklet.java

```diff
@@ -0,0 +1,235 @@
+import org.springframework.batch.core.step.tasklet.Tasklet;
+import org.springframework.batch.item.ExecutionContext;
+import org.springframework.batch.repeat.RepeatStatus;
+
+import uk.ac.ebi.eva.accession.pipeline.policies.InvalidVariantSkipPolicy;
+import uk.ac.ebi.eva.commons.batch.io.VcfReader;
+import uk.ac.ebi.eva.commons.core.models.factories.exception.NonVariantException;
+import uk.ac.ebi.eva.commons.core.models.pipeline.Variant;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Compares the the original VCF and accession report VCF, and logs the differences.
+ *
+ * Terminology: The original VCF contains variants, and it will be loaded into the 'variantBuffer'. The accession
+ * report VCF contains the accessioned variants, and it will be loaded into the 'accessionBuffer'.
+ *
+ * The high-level idea of the process is filling those 2 buffers, and when the same coordinates appears in both the
+ * variantBuffer and the accessionBuffer, it means that the variant in those coordinates was correctly accessioned,
+ * so it can be removed from both buffers. At the end, only variants without accessions and accessions without
+ * variants should be left in the buffers after both files were completely read.
+ *
+ * The reason why the buffers are needed is that the accession report can be unordered. In order to avoid
+ * having the original VCF completely loaded in the variantBuffer, a buffer size can be configured. However, to grant
+ * correct behaviour in worst-ordering scenario, the accessionBuffer (for yet-unmatched accessions from the accession
+ * report) has to be able to grow indefinitely.
+ *
+ * To perform self-checks, this tasklet provides the maximum size that the accessionBuffer during the execution, and
+ * also provides the number of iterations needed.
+ *
+ */
+public class ReportCheckTasklet implements Tasklet {
+
+    private static final Logger logger = LoggerFactory.getLogger(ReportCheckTasklet.class);
+
+    private static final int DEFAULT_MAX_BUFFER_SIZE = 100000;
+
+    private VcfReader vcfReader;
+
+    private VcfReader reportReader;
+
+    private long maxVariantBufferSize;
+
+    private long maxAccessionBufferSize;
+
+    private long iterations;
+
+    private SkipPolicy skipPolicy;
+
+    public ReportCheckTasklet(VcfReader vcfReader, VcfReader reportReader) {
+        this.vcfReader = vcfReader;
+        this.reportReader = reportReader;
+        this.maxVariantBufferSize = DEFAULT_MAX_BUFFER_SIZE;
+        this.maxAccessionBufferSize = 0;
+        this.iterations = 0;
+        this.skipPolicy = new InvalidVariantSkipPolicy();
+    }
+
+    @Override
+    public RepeatStatus execute(StepContribution stepContribution, ChunkContext chunkContext) throws Exception {
+        Set<Variant> variantBuffer = new HashSet<>();
+        Set<Variant> accessionBuffer = new HashSet<>();
+        vcfReader.open(new ExecutionContext());
+        reportReader.open(new ExecutionContext());
```

Comment:
Maybe `inputReader` and `inputBuffer` instead of `vcfReader` (because the report is also a VCF) and `variantBuffer` (because the report also contains variants).